packages/server/src/Socket.ts

```diff
@@ -36,5 +36,15 @@ export const createSocket = (server: HTTPServer) => {
     socket.on(GameEvent.UpdateState, (roomName: string, gameState: GameState) => {
       updateGameState(io, store, roomName, gameState);
     });
+
+    socket.on(GameTurnEvent.SelectPlayerCard, (roomName: string, activePlayerCardKey: string | null) => {
+      io.in(roomName).emit(GameTurnEvent.SelectedPlayerCard, activePlayerCardKey);
+    });
+    socket.on(GameTurnEvent.SelectTableCards, (roomName: string, activeCardKeysOnTable: string[]) => {
+      io.in(roomName).emit(GameTurnEvent.SelectedTableCards, activeCardKeysOnTable);
+    });
+    socket.on(GameTurnEvent.AnimateCards, (roomName: string, animatedCardKeys: string[]) => {
```

Comment:
Naming-wise, isn't this the other way around? Client says "here are the animate**d** cards, and server orders the clients to animat**e** cards" .

Reply:
Maybe you are right, but to me it's like client that says animate those cards, and server responded as those the cards that needs to be animated :-p I don't really mind, if you have a better naming convention or a better way to handle this I'm happy to change, this seems too much boilerplate to me

Reply:
The client does not really have a say in what the other clients would do though. It's just telling the server what it has done. The server has the _power_ to dictate what its clients would do.

I think the states and names would go over some overhaul at some point, I don't have a clear picture of what the events are going to be like, so happy to leave.